saskatoon/harvest/api.py

```diff
@@ -197,27 +201,85 @@ def list(self, request, *args, **kwargs):
         return Response({'data': response.data})
 
 
-class BeneficiaryViewset(LoginRequiredMixin, viewsets.ModelViewSet):
-    """Beneficiary viewset"""
+class OrganizationRetrieveView(LoginRequiredMixin, generics.RetrieveAPIView):
+    """
+    Organization retrieve view shared between beneficiaries and equipment points.
+    """
 
     permission_classes = [IsPickLeaderOrCoreOrAdmin]
     queryset = Organization.objects.all().order_by('-actor_id')
-    serializer_class = BeneficiarySerializer
+    serializer_class = OrganizationSerializer
     filter_backends = (filters.DjangoFilterBackend,)
     filterset_class = OrganizationFilter
+
+    def retrieve(self, request, format='html', pk=None):
+        self.template_name = 'app/detail_views/organization/view.html'
+
+        pk = self.get_object().pk
+        response = super(OrganizationRetrieveView, self).retrieve(request, pk=pk)
+
+        if format == 'json':
+            return response
+
+        # default request format is html:
+        return Response({'organization': response.data,
+                         'data': Equipment.objects.filter(owner_id=pk)
+                         })
+
+class BeneficiaryListView(LoginRequiredMixin, generics.ListAPIView):
+    """
+    List view for organizations that are beneficiaries.
+    """
+
+    permission_classes = [IsPickLeaderOrCoreOrAdmin]
+    queryset = Organization.objects.all().order_by('-actor_id')
+    serializer_class = OrganizationSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = BeneficiaryFilter
```

Comment:
What was the problem with keeping the BeneficiaryViewset as is? I am open to renaming it `OrganizationViewset` but as we discussed I don't think it needs to be refactored:
- list view used for the "Beneficiary" tab (in reality it is more a `Organization` list page since we are actually passing all organization and filtering `is_beneficiary=true` by default
- detail view is used for the Organization detail page

and then a separate `EquipementPointListView(LoginRequiredMixin, generics.ListAPIView):` (as you did)

Reply:
the same can be said for equipment points - the only difference being the filter and template

i thought we were separating them both into single generics because both list views are hierarchically equal - operating on the same queryset.

rundown of options was:
1. single viewset (org detail, ben list) + eq_point list -> confusing url routes
2. single viewset w/ custom actions for lists -> better urls, but over-engineered
3. separate generics for all -> like (2) with no inheritance

keeping the viewset as-is would make it confusing why the eq_point list is functionally similar with beneficiary, but not included within the viewset e.g. as a custom action

Reply:
> i thought we were separating them both into single generics because both list views are hierarchically equal - operating on the same queryset.

No I disagree, the queryset for the equipment point list is more restrictive, it should only show equipment points (i.e. `org.is_equipment_point=true` )

Reply:
> 1. single viewset (org detail, ben list) + eq_point list -> confusing relationships and url routes

what is confusing about this option exactly? Imo there doesn't need to be a relationship between the `OrganizationViewSet` and the `EquipmentPointListView`. Imo we are simply adding a new route to list equipment point and the rest was working fine (besides the questionable naming)

Reply:
reverted to an `Organization` viewset and kept the `EquipmentPointListView`